scripts/publish.mjs

```diff
@@ -0,0 +1,136 @@
+
+// Rebuild with modified version.ts
+await tsBuild();
+
+// ---- Publish
+await npmPublish('@nangohq/types');
+await bumpPackageFor('@nangohq/types', ['cli', 'frontend', 'nango-yaml', 'node-client', 'runner-sdk', 'providers']);
+await npmInstall();
+
+await npmPublish('@nangohq/nango-yaml');
+await bumpPackageFor('@nangohq/nango-yaml', ['cli']);
+await npmInstall();
+
+await npmPublish('@nangohq/providers');
+await bumpPackageFor('@nangohq/providers', ['runner-sdk']);
+await npmInstall();
+
+await npmPublish('@nangohq/node');
+await bumpPackageFor('@nangohq/node', ['cli', 'runner-sdk']);
+await npmInstall();
+
+await npmPublish('@nangohq/runner-sdk');
+await bumpPackageFor('@nangohq/runner-sdk', ['cli']);
+await npmInstall();
+
+// TODO: to delete maybe, seems unnecessary
+await npmPublish('@nangohq/frontend');
+await bumpPackageFor('webapp', ['frontend']);
+await npmInstall();
+
+if (!skipCli) {
+    await npmPublish('nango');
+    await npmInstall();
+}
+// ---- /Publish
+
+await $`npm version "${nextVersion}" --no-git-tag-version --allow-same-version`;
+
+echo``;
+await npmInstall();
+echo(chalk.green(`${figures.tick} npm install`));
+echo(chalk.grey('done'));
+
+// Output for post deploy debug
+echo``;
+echo``;
+echo`- git diff`;
+const res = await $`git diff`;
+echo(res.stdout);
+
+async function tsBuild() {
+    await spinner('npm run ts-build', async () => {
+        await $`npm run ts-build`;
+    });
+    echo(chalk.green(`${figures.tick} npm run ts-build`));
+}
+
+async function modifyVersionTs() {
+    await spinner('Modifying version.ts..', async () => {
+        const versionFiles = await glob('packages/*/lib/version.ts');
+        for (const file of versionFiles) {
+            const content = (await fs.readFile(file)).toString();
+            await fs.writeFile(file, content.replace(versionRegex, nextVersion));
+        }
+    });
+    echo(chalk.green(`${figures.tick} Modifying version.ts`));
+}
+
+async function npmPublish(packageName) {
+    await spinner(`Publishing ${packageName}...`, async () => {
+        const res = await $({ quiet: true, nothrow: true })`npm view "${packageName}@${nextVersion}"`;
+        const hasError = res.stderr.includes('npm error code');
+        const hasNotBeenPublished = res.stderr.includes('npm error code E404');
+        if (!hasError && !hasNotBeenPublished) {
+            echo(chalk.blue(`${figures.tick} Not published ${packageName} (already exists)`));
+            return;
+        }
+
+        if (hasError && !hasNotBeenPublished) {
+            echo``;
+            echo`${chalk.red(`An error occurred`)}`;
+            console.log(res);
+            echo`${res.stderr}`;
+            process.exit(1);
+        }
+
+        await $`npm version ${nextVersion} -w "${packageName}"`;
+        await $`npm publish --access public -w "${packageName}"`;
+
+        echo(chalk.green(`${figures.tick} Published ${packageName}      `));
+    });
+}
+
+async function bumpPackageFor(packageName, folders) {
```

Comment:
we could be smarter here and find all the packages.json in `packages` subfolders and do a `replace(new RegExp(`${packageName}": "${versionRegex}"`), `${packageName}": "${nextVersion}"`)`. 
Then we don't have to pass the folders list and modify this script when adding new packages

Reply:
done in 65851d7bc8fbf20785f94a93912b2ba2dc26c88c